### DISCUSSION TITLE
flash the new firmware

### DISCUSSION THREAD

**Discussion by CalvinKoung** [bitcraze/discussions#1194] [Q&A]
Hi, 
Can somebody instruct me how could I flash the crazyflie firmware include lps firmware ? I have a lot of questions relate the flash the firmware. I present below.

1. What is different with platform filter ? (bolt, cf2, flapper, and tag)
2. If I modify the firmware and I want to flash to the carzyflie my source is  chose from file that correct ? But when I browse, I cannot saw the code file. ｗhy ?
3.  Same as second point, If I modify the lps firmware. When I flash to crazyflie. I need to flash all crazyflie firmware or I can just flash lps firmware.
4. How can I check the crazyflie firmware currently ?

Those above questions hope who can responde me or instruct me. Thanks

**Comment by knmcguire** [comment]
hi! Let me try to help out.l

1. This is necessary since there are certain defines and drivers that should be enabled during compile time for that particular platform. 
2. These instructions should be sufficient to explain how to build and flash the crazyflie firmware. https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/building-and-flashing/build/. After building the .elf or .bin files are in the build folder. 
3. If you changed anything in the LPS code on the crazyflie, you'll need to reflash a completely new crazyflie firmware binary on it. 
4. To see which crazyflie-firmware version you have flashed on your current crazyflie, the console tab on the cfclient can help with that.

**Reply by CalvinKoung** [reply]
ok, got it. I try to do. thanks

**Reply by CalvinKoung** [reply]
@knmcguire Hi, I followed the step by step offical firmware update in crazyflie, but encounter a some problems. I always saw the status is idle and nothing change in my crazyflie ?

**Reply by knmcguire** [reply]
Hi!

Have you tried the terminal based `make cload` or `cfloader` or are you trying to flash from the cfclient? Because it sounds like you are trying to do the latter, while the instructions I gave were terminal based for flashing new and modified firmware

**Reply by CalvinKoung** [reply]
I use the cfclient and followed this instruction
https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/

**Reply by knmcguire** [reply]
Alright, got it. If you flash adjusted firmware, the best way is to use the flashing instructions of the crazyflie firmware itself: 
- To compile: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/building-and-flashing/build/#compiling
- To flash: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/building-and-flashing/build/#make-targets

**Reply by CalvinKoung** [reply]
Sorry let you misunderstood, I flash the firmaware from offical not adjust from me.

**Reply by ataffanel** [reply]
This might be due to an incompatibility of older client with the new firmwares. Firmware version 2024.2+ can only be flashed with clients 2024.2+. If you have installed your client with pip, you can update it with "pip install --upgrade cfclient".

**Reply by CalvinKoung** [reply]
but I saw the latest version is 2023.10. It is correct ? https://github.com/bitcraze/bitcraze-vm/releases

**Reply by knmcguire** [reply]
@ataffanel is talking about the latest version of the python client: https://github.com/bitcraze/crazyflie-clients-python. The VM version and the client version are not related to eachother, so within the VM you can update the cfclient to the latest version of 2024.2

**Reply by CalvinKoung** [reply]
ok, I completed flash the offical firmware (2024.2) today ~ thanks
